REGRESSION (15.4.x): Status bar loses its color after PDF download

dcrousso opened the pull request:
#### 89cbb8ee6ca0ac4237096199725f10e09e785386
<pre>
REGRESSION (15.4.x): Status bar loses its color after PDF download
<a href="https://bugs.webkit.org/show_bug.cgi?id=239898">https://bugs.webkit.org/show_bug.cgi?id=239898</a>
&lt;rdar://problem/92544725&gt;

Reviewed by Tim Horton.

* Source/WebKit/UIProcess/ios/WKPDFView.mm:
(-[WKPDFView web_initWithFrame:webView:mimeType:]):
243980@main changed `WKScrollView` to have a `-[WKScrollView _setBackgroundColorInternal:]` in order
to know whether the `backgroundColor` is being set by WebKit or by the owner of the `WKWebView`. The
former allows for WebKit to continue automatically updating the `backgroundColor` based on the page
content, whereas the latter means WebKit should not as the client explicitly asked for that color.
Since `WKPDFView` is an internal implementation detail of WebKit, it should fall into the former.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _wkScrollView]): Added.
(-[WKWebView _updateScrollViewBackground]):
* Source/WebKit/UIProcess/ios/WKScrollView.mm:
(-[WKScrollView _setBackgroundColorInternal:]):
Expose a way for internal logic to access the `WKScrollView` from the `WKWebView`.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm:
(TEST.WKPDFView.BackgroundColor): Added.
* Tools/TestWebKitAPI/Tests/WebKit/red.html: Added.
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/253372@main">https://commits.webkit.org/253372@main</a>
</pre>

hortont424 commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm:
```diff
@@ -172,6 +172,30 @@ static void createHostViewForExtensionIdentifier(void(^callback)(id hostViewCont
+    auto redColor = adoptCF(CGColorCreate(sRGBColorSpace.get(), redColorComponents));
+
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600)]);
+    NSLog(@"<DR> %@", [webView scrollView].backgroundColor);
```
no `<DR>` please

dcrousso replied:
oh lol oops

hortont424 commented on Tools/TestWebKitAPI/Tests/WebKit/red.html:
```diff
@@ -0,0 +1 @@
+<body style="background-color: red"></body>
```
Could just be inline in the test, but this is fine too

dcrousso replied:
i tried doing that but it didnt seem to work correctly with `goBack`

dcrousso committed:
REGRESSION (15.4.x): Status bar loses its color after PDF download
https://bugs.webkit.org/show_bug.cgi?id=239898
<rdar://problem/92544725>

Reviewed by Tim Horton.

* Source/WebKit/UIProcess/ios/WKPDFView.mm:
(-[WKPDFView web_initWithFrame:webView:mimeType:]):
243980@main changed `WKScrollView` to have a `-[WKScrollView _setBackgroundColorInternal:]` in order
to know whether the `backgroundColor` is being set by WebKit or by the owner of the `WKWebView`. The
former allows for WebKit to continue automatically updating the `backgroundColor` based on the page
content, whereas the latter means WebKit should not as the client explicitly asked for that color.
Since `WKPDFView` is an internal implementation detail of WebKit, it should fall into the former.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _wkScrollView]): Added.
(-[WKWebView _updateScrollViewBackground]):
* Source/WebKit/UIProcess/ios/WKScrollView.mm:
(-[WKScrollView _setBackgroundColorInternal:]):
Expose a way for internal logic to access the `WKScrollView` from the `WKWebView`.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPDFView.mm:
(TEST.WKPDFView.BackgroundColor): Added.
* Tools/TestWebKitAPI/Tests/WebKit/red.html: Added.
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/253372@main
